d3a_interface/sim_results/kpi.py

```diff
@@ -147,26 +148,25 @@ def update_area_kpi(self, area_dict, core_stats):
 
 class SavingsKPI:
     def __init__(self):
-        self.producer_ess_set = set()
-        self.consumer_ess_set = set()
-        self.house_list = list()
-        self.fit_revenue = 0.
-        self.utility_bill = 0.
-        self.base_case_revenue = 0.
-        self.d3a_revenue = 0.
-        self.saving_absolute = 0.
-        self.saving_percentage = 0.
-
-    def calculate_saving_kpi(self, area_dict, core_stats, gf_alp):
-        """
+        self.producer_ess_set = set()  # keeps set of house's producing/ess devices
+        self.consumer_ess_set = set()  # keeps set of house's consuming/ess devices
+        self.fit_revenue = 0.  # revenue achieved by producing selling energy via FIT scheme
+        self.utility_bill = 0.  # cost of energy purchase from energy supplier
+        self.base_case_revenue = 0.  # standard revenue of a house participating in FIT scheme
+        self.d3a_revenue = 0.  # standard revenue of a house participating in D3A
+        self.saving_absolute = 0.  # savings achieved by a house via participating in D3A
+        self.saving_percentage = 0.  # savings in percentage wrt FIT via participating in D3A
+
+    def calculate_savings_kpi(self, area_dict, core_stats, gf_alp):
+        """Calculates the referenced saving from feed-in tariff based participation vs D3A
         Args:
             area_dict: contain nested area info
-            core_stats: contain area's raw/key statistics
+            core_stats: contain area"s raw/key statistics
```

Comment:
Nit: a side-effect of the global replacement of `'` :)

Reply:
reminder